src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -152,8 +152,8 @@ private async Task<AuthenticationResult> SendTokenRequestForManagedIdentityAsync
 
             await ResolveAuthorityAsync().ConfigureAwait(false);
 
-            ManagedIdentityClient managedIdentityClient = 
-                new ManagedIdentityClient(AuthenticationRequestParameters.RequestContext);
+            ManagedIdentityClient managedIdentityClient = await ManagedIdentityClient.CreateAsync(
```

Comment:
So far, I only see the probe logic. Did you consider working directly on Robbie's branch which has the same?

Reply:
This is just a prototype, not the final code Robbie is writing